libs/sdk-backend-bear/src/backend/workspace/dashboards/index.ts

```diff
@@ -329,13 +330,25 @@ export class BearWorkspaceDashboards implements IWorkspaceDashboardsService {
             ? await this.getScheduledMailObjectLinksForDashboardAndCurrentUser(dashboardRef)
             : await this.getScheduledMailObjectLinksForDashboard(dashboardRef);
 
-        const userMap: Map<string, IUser> = options.loadUserData
-            ? await updateUserMap(
-                  new Map(),
-                  compact(flatMap(scheduledMailObjectLinks, (link) => [link.author, link.contributor])),
-                  this.authCall,
-              )
-            : new Map();
+        let userMap: Map<string, IUser> = new Map();
+        if (options.loadUserData) {
+            userMap = await updateUserMap(
+                userMap,
+                compact(flatMap(scheduledMailObjectLinks, (link) => [link.author, link.contributor])),
+                this.authCall,
+            );
+
+            // if listing users is not allowed add at least the current user
+            if (userMap.values.length === 0) {
+                await this.authCall(async (sdk) => {
+                    const profile = await sdk.user.getCurrentProfile();
+                    const user = convertUser(profile);
+                    if (profile.links?.self) {
+                        userMap.set(profile.links?.self, user);
+                    }
```

Comment:
This may be simplified by using `userUriFromAuthenticatedPrincipalWithAnonymous` which should already return user uri. Check other usages in this file.

Reply:
here we need the whole `IUser` but the suggested function provides just URI